pkg/providers/vsphere/upgrade/virtualmachine/vm_schema_upgrade.go

```diff
@@ -516,50 +684,46 @@ func reconcileCdromSpec(
 	bFileNameToCdromInfo map[string]pkgutil.VirtualCdromInfo,
 	ctrlKeyToCtrlStatus map[int32]vmopv1.VirtualControllerStatus) {
 
-	logger := pkglog.FromContextOrDefault(ctx)
+	logger := pkglog.FromContextOrDefault(ctx).WithValues("name", spec.Name)
+
+	if !needsCdromBackfill(spec) {
+		logger.V(4).Info("Skipping CD-ROM due to no placement fields to backfill")
+		return
+	}
 
 	bFileName, err := virtualmachine.GetBackingFileNameByImageRef(
 		ctx, k8sClient, spec.Image, vm.Namespace, false, nil)
 	if err != nil {
-		logger.Error(err,
-			"Error getting backing file name", "spec", spec)
+		logger.Error(err, "Error getting backing file name")
```

Comment:
Nit: maybe include "CD-ROM" in this log message too like the others here so context is more obvious (depending on how descriptive spec name is)